lib/oauth_client.rb

```diff
@@ -0,0 +1,138 @@
+# frozen_string_literal: true
+module Zoom
+  class OAuthClient
+    OAUT_URL = "https://zoom.us/oauth/token"
+
+    def initialize(api_url, end_point, authorization = nil)
+      @api_url = api_url
+      @end_point = end_point
+      @max_tries = 5
+      @tries = 0
+
+      if authorization
+        @authorization = authorization
+      else
+        @oauth = true
+        @authorization =
+          (
+            if SiteSetting.s2s_oauth_token.empty?
+              get_oauth
+            else
+              SiteSetting.s2s_oauth_token
+            end
+          )
+      end
+    end
+
+    def get
+      response =
+        Excon.get(
+          "#{@api_url}#{@end_point}",
+          headers: {
+            Authorization: "Bearer #{@authorization}"
+          }
+        )
+      response.body =
+        JSON.parse(
+          response.body,
+          symbolize_names: true
+        ) unless response.body.blank?
+
+      if ([400, 401].include? response.status) && @tries < @max_tries
```

Comment:
This logic is repeated in `post` and `deleted` so it's probably worth moving to a method and calling from each of these